lighthouse-core/audits/load-fast-enough-for-pwa.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+/** @fileoverview
+ *  This audit evaluates if a page's load performance is fast enough for it to be considered a PWA.
+ *  We are doublechecking that the network requests were throttled (or slow on their own)
+ *  Afterwards, we report if the TTI is less than 10 seconds.
+ */
+
+const Audit = require('./audit');
+const TTIMetric = require('./time-to-interactive');
+const Emulation = require('../lib/emulation');
+
+const Formatter = require('../report/formatter');
+
+// Maximum TTI to be considered "fast" for PWA baseline checklist
+//   https://developers.google.com/web/progressive-web-apps/checklist
+const MAXIMUM_TTI = 10 * 1000;
+
+class LoadFastEnough4Pwa extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'PWA',
+      name: 'load-fast-enough-for-pwa',
+      description: 'Page load is fast enough on 3G',
+      helpText: 'Satisfied if the _Time To Interactive_ duration is shorter than _10 seconds_, as defined by the [PWA Baseline Checklist](https://developers.google.com/web/progressive-web-apps/checklist).',
+      requiredArtifacts: ['traces', 'networkRecords']
+    };
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!AuditResult}
+   */
+  static audit(artifacts) {
+    const networkRecords = artifacts.networkRecords[Audit.DEFAULT_PASS];
+    const allRequestLatencies = networkRecords.map(record => {
+      if (!record._timing) return Infinity;
```

Comment:
because if there is no timing data on a request then we don't care about it. we don't want to consider it for this analysis.

i'm also happy to filter netRecords to only the ones with timing first.